airbyte-metrics/reporter/src/main/java/io/airbyte/metrics/reporter/MetricRepository.java

```diff
@@ -140,6 +140,66 @@ having count(*) < 1440 / cast(c.schedule::jsonb->'units' as integer)
         + ctx.fetchOne(queryForAbnormalSyncInMinutesInLastDay).get("cnt", long.class);
   }
 
+  long numberOfJobsRunningUnusuallyLong() {
+    // Definition of unusually long means runtime is more than 2x historic avg run time or 15
+    // minutes more than avg run time, whichever is greater.
+    final var query =
+        """
+        -- pick average running time and last sync running time in attempts table.
+          select
+            current_running_attempts.connection_id,
+            current_running_attempts.running_time,
+            historic_avg_running_attempts.avg_run_sec
+            from
+            	(
+             -- Sub-query-1: query the currently running attempt's running time.
+                (
+                  select
+                    jobs.scope as connection_id,
+                    extract(epoch
+                  from
+                    age(NOW(), attempts.created_at)) as running_time
```

Comment:
when i started working on this, I remember the attempts table not being [in a great state](https://airbytehq-team.slack.com/archives/C02TXQ020QM/p1663219967719159).

Want to confirm this pulls the latest running attempt for the connection.

Reply:
Yeah the problem still exists, but coupling attempts status with jobs status could eliminate the problem:

`select  count(*) from attempts inner join jobs on jobs.id = attempts.job_id where attempts.status = 'running' and jobs.status = 'running';` yields 62 result which is much more reasonable